Explorer/Assets/DCL/CharacterMotion/Systems/FeetIKSystem.cs

```diff
@@ -0,0 +1,252 @@
+        private readonly ElementBinding<float> ikWeightChangeSpeed;
+        private readonly ElementBinding<float> ikPositionChangeSpeed;
+        private readonly ElementBinding<float> ikDistance;
+        private readonly ElementBinding<float> spherecastWidth;
+        private readonly ElementBinding<float> twistLimitX;
+        private readonly ElementBinding<float> twistLimitY;
+        private SingleInstanceEntity settingsEntity;
+        private bool isInitialized;
+
+        private FeetIKSystem(World world, IDebugContainerBuilder debugBuilder) : base(world)
+        {
+            debugBuilder.AddWidget("Locomotion: Feet IK")
+                        .AddSingleButton("Toggle Enable", () => disableWasToggled = true)
+                        .AddFloatField("IK Change Speed", ikWeightChangeSpeed = new ElementBinding<float>(0))
+                        .AddFloatField("IK Position Speed", ikPositionChangeSpeed = new ElementBinding<float>(0))
+                        .AddFloatField("IK Distance", ikDistance = new ElementBinding<float>(0))
+                        .AddFloatField("Spherecast Width", spherecastWidth = new ElementBinding<float>(0))
+                        .AddFloatField("Twist Limit X", twistLimitX = new ElementBinding<float>(0))
+                        .AddFloatField("Twist Limit Y", twistLimitY = new ElementBinding<float>(0));
+        }
+
+        public override void Initialize()
+        {
+            settingsEntity = World.CacheCharacterSettings();
+        }
+
+        protected override void Update(float t)
+        {
+            ICharacterControllerSettings settings = settingsEntity.GetCharacterSettings(World);
+
+            Vector2 twistLimits = settings.FeetIKTwistAngleLimits;
+
+            if (!isInitialized)
+            {
+                isInitialized = true;
+                ikWeightChangeSpeed.Value = settings.IKWeightSpeed;
+                ikPositionChangeSpeed.Value = settings.IKPositionSpeed;
+                ikDistance.Value = settings.FeetIKHipsPullMaxDistance;
+                spherecastWidth.Value = settings.FeetIKSphereSize;
+                twistLimitX.Value = twistLimits.x;
+                twistLimitY.Value = twistLimits.y;
+            }
+
+            settings.IKWeightSpeed = ikWeightChangeSpeed.Value;
+            settings.IKPositionSpeed = ikPositionChangeSpeed.Value;
+            settings.FeetIKHipsPullMaxDistance = ikDistance.Value;
+            settings.FeetIKSphereSize = spherecastWidth.Value;
+            twistLimits.x = twistLimitX.Value;
+            twistLimits.y = twistLimitY.Value;
+            settings.FeetIKTwistAngleLimits = twistLimits;
+
+            UpdateIKQuery(World, t);
+        }
+
+        [Query]
+        private void UpdateIK(
+            [Data] float dt,
+            ref FeetIKComponent feetIKComponent,
+            ref AvatarBase avatarBase,
+            in CharacterRigidTransform rigidTransform,
+            in ICharacterControllerSettings settings,
```

Comment:
How does IK work for other avatars considering `settings` are required? Do we apply this settings for other players as well?

Reply:
Yeah those settings are pretty global for all locomotion related systems.